api/v1beta1/argocd_types.go

```diff
@@ -162,6 +169,13 @@ type ArgoCDApplicationSet struct {
 
 	// SCMRootCAConfigMap is the name of the config map that stores the Gitlab SCM Provider's TLS certificate which will be mounted on the ApplicationSet Controller (optional).
 	SCMRootCAConfigMap string `json:"scmRootCAConfigMap,omitempty"`
+
+	// Enabled is the flag to enable the Application Set Controller during ArgoCD installation. (optional, default `true`)
+	Enabled *bool `json:"enabled,omitempty"`
```

Comment:
This seems unnecessary for appset, since we can already enable/disable appset by leaving it as nil or not nil
is there some way this field will become useful in the future?

Reply:
Tried keeping it consistent with the format being used for other core components of ArgoCD. It would be better to update that behavior eventually as ApplicationSet is considered a core component for ArgoCD now.